frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -570,6 +574,78 @@ pub mod pallet {
             Ok(().into())
         }
 
+        /// Lock up and stake unbonded chunks of origin account.
+        /// 
+        /// `value` must be more than the `minimum_balance` specified by `MinimumStakingAmount`
+        /// unless account already has bonded value equal or more than 'minimum_balance'.
+        /// 
+        /// The dispatch origin for this call must be _Signed_ by the staker's account.
+        #[pallet::weight(T::WeightInfo::rebond_and_stake())]
+        pub fn rebond_and_stake(
+            origin: OriginFor<T>,
+            contract_id: T::SmartContract,
+            #[pallet::compact] value: BalanceOf<T>,
```

Comment:
In order to save you some time with dev & test work - this should be removed.
The call should be kept as simple as possible - no sorting and such stuff since it doesn't provide any real value to the user.

They can only rebond ALL chunks.

Reply:
Thank you so much for notifying!